### DISCUSSION TITLE
Option to toggle autocomplete?

### DISCUSSION THREAD

**Discussion by Reefersleep** [LazyVim/LazyVim#3302] [Q&A]
Is there a way to turn autocompletion (which I guess happens via nvim-cmp) on and off with a toggle?
Or could there be? :) 

If I'm writing, say, a docstring, with autocompletion on, the autcomplete list is blinking in and out of view all while I'm writing, which is very distracting. But more seriously, if I hit what I consider the end of a line and press Enter to go to a new line, I risk inserting the first item in the autocomplete list, which is confusing and inefficient, to say the least.

Just the option to toggle autocompletion off while writing, then turn it on afterwards, would work well for me.

I did read https://github.com/LazyVim/LazyVim/discussions/2885 , in which the author has a similar issue and it is resolved by turning off automatic autocompletion and invoking it manually with `<C-Space>`, and this will have to do for me for now, too. I wonder if the toggle I describe is possible, and whether it might be interesting to others in the community :)

**Comment by dpetka2001** [comment]
Related #2981 that didn't gain much traction and was closed, since the obvious solution is to just use the setting `nvim-cmp` provides to disable auto-completion and then manually invoke completion with whatever key you have set (you can change that if you don't like the default LazyVim `<Cr>` to something else). 

Also related https://github.com/hrsh7th/nvim-cmp/issues/261 that `nvim-cmp` also didn't implement and left it up to the users to come up with something in their personal configuration if they wish so. There are some recipes in that thread which you could try out in your personal configuration.

**Comment by Dronakurl** [comment]
Here is my cmp config which toggles autocomplete with `<leader>ua`:

```lua
return {
  {
    "hrsh7th/nvim-cmp",
    init = function()
      vim.g.cmp_disabled = false
    end,
    opts = function(_, opts)
      opts.enabled = function()
        -- local context = require("cmp.config.context")
        if vim.g.cmp_disabled == true then
          return false
        end
        -- some other conditions (like not in commments) can go here
        return not disabled
      end
    end,
    keys = {
      {
        "<leader>ua",
        function()
          vim.g.cmp_disabled = not vim.g.cmp_disabled
          local msg = ""
          if vim.g.cmp_disabled == true then
            msg = "Autocompletion (cmp) disabled"
          else
            msg = "Autocompletion (cmp) enabled"
          end
          vim.notify(msg, vim.log.levels.INFO)
        end,
        noremap = true,
        silent = true,
        desc = "toggle autocompletion",
      },
    },
  },
}
```
I think, this does what is asked for

**Comment by b11e** [comment]
has anyone got this working with blink-cmp?

**Reply by b11e** [reply]
i set up a discussion since I could not get a working example
https://github.com/Saghen/blink.cmp/discussions/1173#discussion-7929919

**Reply by dpetka2001** [reply]
https://github.com/LazyVim/LazyVim/discussions/5186#discussioncomment-12090959

**Reply by b11e** [reply]
thank you

> [#5186 (comment)](https://github.com/LazyVim/LazyVim/discussions/5186#discussioncomment-12090959)
